alerting/src/main/kotlin/org/opensearch/alerting/DocumentLevelMonitorRunner.kt

```diff
@@ -86,28 +91,39 @@ object DocumentLevelMonitorRunner : MonitorRunner() {
         val index = docLevelMonitorInput.indices[0]
         val queries: List<DocLevelQuery> = docLevelMonitorInput.queries
 
+        val monitorMetadata = if (monitor.metadataId.isNullOrEmpty()) createMonitorMetadata(monitor.id)
+        else {
+            try {
+                getMonitorMetadata(monitorCtx.client!!, monitorCtx.xContentRegistry!!, monitor.metadataId)
+            } catch (e: Exception) {
+                logger.warn("Monitor metadata with id, ${monitor.metadataId}, is missing for monitor: ${monitor.id}")
+                createMonitorMetadata(monitor.id)
+            }
```

Comment:
This is not idempotent as read failures due to multiple reason can result into conflicting write operations? Given monitor metadata has critical state information, such as lastRunContext, we should rather bail out and surface the error.

Reply:
I refined the exceptions to look for config index not existing and if it could not find the result.